fem/fespace.hpp

```diff
@@ -1371,12 +1458,22 @@ class FiniteElementSpace
        Safe to call multiple times, does nothing if space already up to date. */
    virtual void Update(bool want_transform = true);
 
+   /** P-refine and update the space. If @a want_transfer, also maintain the old
+       space and a transfer operator accessible by GetPrefUpdateOperator(). */
+   virtual void PRefineAndUpdate(const Array<pRefinement> & refs,
+                                 bool want_transfer = true);
+
+   /// Return true iff p-refinement is supported in this space.
+   bool PRefinementSupported();
+
    /// Get the GridFunction update operator.
    const Operator* GetUpdateOperator() { Update(); return Th.Ptr(); }
 
    /// Return the update operator in the given OperatorHandle, @a T.
    void GetUpdateOperator(OperatorHandle &T) { T = Th; }
 
+   std::shared_ptr<const PRefinementTransferOperator> GetPrefUpdateOperator();
```

Comment:
Document

Reply:
Done in fc63c5e90b5ace705c64319123d3715d1dbe250f